core/src/subject/SubjectEntity.ts

```diff
@@ -0,0 +1,213 @@
+      let resolveLanguage
+      if (resolveLanguageResults && resolveLanguageResults.length > 0) {
+        resolveLanguage = resolveLanguageResults[0].Language
+      }
+
+      if (resolveLanguage) {
+        value = await this.#perspective.createExpression(value, resolveLanguage)
+      }
+      await this.#perspective.executeAction(actions, this.#baseExpression, [{ name: "value", value }])
+    }
+  }
+
+  private async setCollectionSetter(key: string, value: any) {
+    let collectionSetters = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), collection_setter(C, "${singularToPlural(key)}", Setter)`)
+    if (!collectionSetters) collectionSetters = []
+
+    if (collectionSetters.length > 0) {
+      const actions = eval(collectionSetters[0].Setter)
+
+      if (Array.isArray(value)) {
+        await this.#perspective.executeAction(actions, this.#baseExpression, value.map(v => ({ name: "value", value: v })))
+      } else {
+        await this.#perspective.executeAction(actions, this.#baseExpression, [{ name: "value", value }])
+      }
+    }
+  }
+
+  private async setCollectionAdder(key: string, value: any) {
+    let adders = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), collection_adder(C, "${singularToPlural(key)}", Adder)`)
+    if (!adders) adders = []
+
+    if (adders.length > 0) {
+      const actions = eval(adders[0].Adder)
+      if (Array.isArray(value)) {
+        await Promise.all(value.map(v => this.#perspective.executeAction(actions, this.#baseExpression, [{ name: "value", value: v }])))
+      } else {
+        await this.#perspective.executeAction(actions, this.#baseExpression, [{ name: "value", value }])
+      }
+    }
+  }
+
+  async save() {
+    this.#subjectClass = await this.#perspective.stringOrTemplateObjectToSubjectClass(this)
+    
+    await this.#perspective.createSubject(this, this.#baseExpression);
+
+    await this.update()
+  }
+
+  async update() {
+    this.#subjectClass = await this.#perspective.stringOrTemplateObjectToSubjectClass(this)
+
+    const entries = Object.entries(this);
+
+    for (const [key, value] of entries) {
+      if (value) {
+        await this.setProperty(key, value);
+    
+        if (value?.action) {
+          switch (value.action) {
+            case 'setter':
+              await this.setCollectionSetter(key, value.value)
+              break;
+            case "adder":
+              await this.setCollectionAdder(key, value.value)
+              break;
+            default:
+              await this.setCollectionSetter(key, value.value)
+              break;
+          }
+        } else {
+          await this.setCollectionSetter(key, value)
+        }
+      }
+    }
+  }
+
+  async get() {
+    this.#subjectClass = await this.#perspective.stringOrTemplateObjectToSubjectClass(this)
+    
+    return await this.getData()
+  }
+
+  // TODO: implement simple quering like limit, skip etc.
+  async find() {
```

Comment:
Find it weird that this is an instance method, and not a static class method. I understand that we need a PerspectiveProxy to do anything, but we can provide that as argument. Clearer interface than having to create an instance which is not used as an instance of the Subject Class..
```suggestion
  static async all(perspective: PerspectiveProxy) {
```